### PR TITLE
#2373 feature add: Vertical Slice Architecture.

### PR DESCRIPTION
**Pull request by sugan0tech**
Added the Vertical Slice Architecture #2373
 ̑

### CODE REVIEW

File: vertical-slice-architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: Vertical-Slice-Architecture
```
Comment (iluwatar) [inline vertical-slice-architecture/README.md:2]:
```suggestion
title: Vertical Slice Architecture
```

File: vertical-slice-architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+aka: Layer-By-Feature
+category: Architectural
+language: en
+tag:
+- Decoupling
+---
+
+## Intent
+
+package the application based on features. Each feature will have its own set of layers (
```
Comment (iluwatar) [inline vertical-slice-architecture/README.md:12]:
```suggestion
Package the application based on features. Each feature will have its own set of layers (
```

File: vertical-slice-architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+> With vertical slice architecture we can have high cohesion within package and low coupling
+> among the packages. In Conceptual term
+
+> Consider that you are going to make a backend service for a online e-commerce application.
+> initially you make it with usual grouping of controllers, models etc. but as the application
+> grows more it's requires implementation of new features. Let's say that you thought of having
+> orders, customers and products associated layers. But now you need to include another set of
+> features with Cart system and wishlists. Now it's really hard to integrate those features it
+> requires lot's of dependency modifications and mocking. So if you make the package by feature
+> it will be really feasible for future additions. General example.
```
Comment (iluwatar) [inline vertical-slice-architecture/README.md:26]: For the explanation, please use the standard headings as described here: https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template Check the other patterns for examples as well.

File: vertical-slice-architecture/src/main/java/com/iluwatar/verticalslicearchitecture/App.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * Main application.
+ * The main objective of this code sample is for you to have a look at how the view, entity, repository and Service are
+ * organized.
+ * No matter what application you are building Layer by Feature will always give the upper hand of better
+ * maintainability.
+ */
```
Comment (iluwatar) [inline vertical-slice-architecture/src/main/java/com/iluwatar/verticalslicearchitecture/App.java:36]: In this section, please describe the pattern briefly. Then explain how the example implements the pattern.

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: vertical-slice-architecture/README.md (hunk 3 above)
Comment (sugan0tech) [reply vertical-slice-architecture/README.md]: Updated, Do I need to have a code sample for this?.

**Comment by sugan0tech** [comment]
@iluwatar  updated the changes. can you check it out?

**Pull request closed without merging**